server/app/views/admin/programs/ProgramBlockPredicateConfigureView.java

```diff
@@ -0,0 +1,621 @@
+    } else {
+      valueRowContainer.with(
+          renderValueRow(questionDefinitions, 1, /* maybeAndNode= */ Optional.empty()));
+    }
+
+    formTag.with(valueRowContainer);
+    formTag.with(div(makeSvgTextButton("Add values", Icons.ADD).withId("predicate-add-value-row")));
+    formTag.with(submitButton("Save condition").withClasses("my-4"));
+
+    return formTag;
+  }
+
+  private static DivTag renderConfiguratorEligibilityHeader() {
+    return div(
+            text("Applicant is eligible when:"),
+            input()
+                .withType("hidden")
+                .withName("predicateAction")
+                .withValue(PredicateAction.ELIGIBLE_BLOCK.name()))
+        .withClasses("font-bold");
+  }
+
+  private static DivTag renderConfiguratorVisibilityHeader(
+      Optional<PredicateDefinition> maybeExistingPredicate) {
+    DivTag container =
+        div("Screen is ").withClasses("font-bold", ReferenceClasses.PREDICATE_ACTION);
+    SelectTag selectTag = select().withName("predicateAction");
+
+    ImmutableList.of(PredicateAction.HIDE_BLOCK, PredicateAction.SHOW_BLOCK)
+        .forEach(
+            action ->
+                selectTag.with(
+                    option(action.toDisplayString())
+                        .withClasses(
+                            ReferenceClasses.MULTI_OPTION_QUESTION_OPTION,
+                            ReferenceClasses.MULTI_OPTION_VALUE)
+                        .withCondSelected(
+                            maybeExistingPredicate
+                                .map(PredicateDefinition::action)
+                                .map(existingAction -> existingAction.equals(action))
+                                .orElse(false))
+                        .withValue(action.name())));
+
+    container.with(selectTag);
+
+    return container;
+  }
+
+  private ImmutableList<PredicateExpressionNode> getExistingAndNodes(
+      PredicateDefinition existingPredicate) {
+    switch (existingPredicate.computePredicateFormat()) {
+      case SINGLE_QUESTION:
+        {
+          return ImmutableList.of(
+              PredicateExpressionNode.create(
+                  AndNode.create(ImmutableList.of(existingPredicate.rootNode()))));
+        }
+
+      case OR_OF_SINGLE_LAYER_ANDS:
+        {
+          return existingPredicate.rootNode().getOrNode().children();
+        }
+
+      default:
+        {
+          throw new RuntimeException(
+              String.format(
+                  "Unrecognized predicate format: %s", existingPredicate.computePredicateFormat()));
+        }
+    }
+  }
+
+  private DivTag renderQuestionHeaders(
+      ImmutableList<QuestionDefinition> questionDefinitions,
+      Optional<PredicateDefinition> maybeExistingPredicate) {
+    DivTag container = div().withClasses("flex", "py-4");
+    int columnNumber = 1;
+
+    if (maybeExistingPredicate.isPresent()) {
+      ImmutableList<LeafOperationExpressionNode> leafNodes =
```

Comment:
Leave a  javadoc that we're arbitrarily ordering by some ID (admin name?) to maintain a stable though arbitrary ordering.

How is the question selection list ordered? Ideally that would also be stable and we use the same pattern so that the two interfaces don't give different orders.

Reply:
using question admin name for ordering now